src/nl/rubensten/texifyidea/inspections/bibtex/BibtexDuplicateBibliographyInspection.kt

```diff
@@ -31,20 +31,33 @@ open class BibtexDuplicateBibliographyInspection : TexifyInspectionBase() {
     override fun inspectFile(file: PsiFile, manager: InspectionManager, isOntheFly: Boolean): MutableList<ProblemDescriptor> {
         val descriptors = descriptorList()
 
+        val groupedIncludes = mutableMapOf<String, MutableList<LatexCommands>>()
+
         LatexIncludesIndex.getItemsInFileSet(file).asSequence()
                 .filter { it.name == "\\bibliography" || it.name == "\\addbibresource" }
-                .groupBy { it.requiredParameters.getOrNull(0) }
-                .filter { it.key != null && it.value.size > 1 }
-                .flatMap { it.value }
                 .forEach {
-                    descriptors.add(manager.createProblemDescriptor(
-                            it,
-                            TextRange(0, it.requiredParameter(0)?.length!!).shiftRight(it.commandToken.textLength + 1),
-                            "Bibliography file is included multiple times",
-                            ProblemHighlightType.GENERIC_ERROR,
-                            isOntheFly,
-                            RemoveOtherCommandsFix(it.requiredParameter(0)!!)
-                    ))
+                    for (fileName in it.includedFileNames() ?: emptyList()) {
+                        groupedIncludes.getOrPut(fileName) { mutableListOf() }.add(it)
+                    }
+                }
+
+        groupedIncludes.asSequence()
+                .filter { it.value.size > 1 }
+                .forEach {
```

Comment:
```suggestion
                .forEach { (fileName, commands) ->
```
And of course account for `distinct()`. Maybe `HashSet` (or when order is important; `LinkedHashSet`) is more suitable for `MutableList` on line 23.

Reply:
True, however `HashSet` is not applicable, because `\bibliography{kameel,kameel}` still needs to trigger. Thus the command must be recorded in the list twice initially, but must only be registered a single problem description, Will change